jsonschema/tests/test_cli.py

```diff
@@ -683,6 +684,73 @@ def test_successful_validation_of_just_the_schema_pretty_output(self):
             stderr="",
         )
 
+    def test_successful_validate_with_specifying_base_uri_absolute_path(self):
+        absolute_path = os.getcwd()
+        try:
+            schema_file = tempfile.NamedTemporaryFile(
+                mode='w+',
+                prefix='schema',
+                suffix='.json',
+                dir=absolute_path,
+                delete=False
+            )
+            self.addCleanup(os.remove, schema_file.name)
+            schema = """
+                    {"type": "object", "properties": {"KEY1":
```

Comment:
Can we simplify the schema we're using to test this behavior? It's a bit hard to follow what's happening with all the string formatting. (I don't understand why we need to format in `"\""`, but even more than that is a bit hard to follow).

We should in theory be able to test this behavior by having a schema with one single validator, a `$ref` with a relative reference, and then we ensure that that relative reference was used when resolving -- e.g. say, a temporary directory with two files, bar with `{"$ref": "foo.json#definitions/num"}` and then another, `foo.json`, with `{"definitions": {"num": {"type": "integer"}}`, and then setting `--base-uri` to the enclosing directory already does it, no?

That'd seem a bit clearer, but if I'm just not seeing how this test is working maybe if you remove some of the unneeded properties in this schema it'd be clearer.